### DISCUSSION TITLE
Game doesn't compile in my existing PIXI project but works fine in a new project

### DISCUSSION THREAD

**Discussion by Giacom** [jnsmalm/pixi3d#172] [Q&A]
I've been getting these compile errors when trying to use PIXI 3D.

I am just doing this:

```ts
import { Container } from "@pixi/display";
import { Container3D } from "pixi3d/pixi7";

let test = new Container();
test.addChild(new Container3D());
```

And I get these compile errors.

```
src/index.ts:43:15 - error TS2345: Argument of type 'Container3D' is not assignable to parameter of type 'DisplayObject'.
  The types of 'transform.position.normalize(...).magnitude' are incompatible between these types.
    Type 'number' is not assignable to type '() => number'.

43 test.addChild(new Container3D());
```

For some reason I only get this error on my existing PIXI project, I tried using your template project but I didn't receive any error about DisplayObject, any clue why that is?

I'm using the same PIXI.js and PIXI-3D.js versions for both the existing and new project.

**Comment by jnsmalm** [comment]
Could it be any difference between the TypeScript config files?

**Reply by Giacom** [reply]
Yeah that's what I first thought but I tried using the same config as the existing project but no luck.

```json5
{
    "compilerOptions": {
        "outDir": "./dist/",
        "target": "es2015",
        "module": "es2015",
        "allowJs": true,
        "declaration": false,
        "sourceMap": true,
        "removeComments": true,
        "isolatedModules": true,
        "moduleResolution": "node",
        "esModuleInterop": true,
        "skipLibCheck": true 
    },

    "include": ["src/**/*.ts"],
    "exclude": ["dist/**"]
}
```

They're also both using the same version of `tsc`.

```bash
$ npx tsc --version
Version 5.0.3
```

**Comment by Giacom** [comment]
I narrowed down the problem.

In my existing project I am using @pixi/math-extra, which I import in a handmade bundle PIXI.ts file that I use.
The new project didn't import this package so it didn't complain and when I do import the package it then produces the compile errors.

I'll create an issue for this.
